hashintel/hash
# Cleanup temporary files when test fail

## 🌟 What is the purpose of this PR?

When tests fail they sometimes leave temporary files in the engine directory.
With this PR there shouldn't be any temporary files left.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1199548034582004/1202206648648212/f) _(internal)_

## 🔍 What does this change?

- Remove "run-{experiment_id}" files
- Wait 1s before cleaning up temporary files so python socket files have time to be created before cleanup

## 📜 Does this require a change to the docs?

No

## ❓ How to test this?

1.  Checkout the branch / view the deployment

// engine library
2.  Try to add a `panic!()` in the engine (for example in this function https://github.com/hashintel/hash/blob/925d5af5b1ac5007f54daf5e86a22200a8f3da0c/packages/engine/src/worker/runner/javascript/mod.rs#L315)
3.  Run `cargo test`
4.  Confirm that no temporary files are left when `cargo test` is done

// JS throw
2.  Try to add a `throw new Error()` in a behavior (for example in this behavior https://github.com/hashintel/hash/blob/925d5af5b1ac5007f54daf5e86a22200a8f3da0c/packages/engine/tests/examples/boids_3d/src/behaviors/create_flock.js#L5)
3.  Run `cargo test`
4.  Confirm that no temporary files are left when `cargo test` is done

// test panic
2.  Try to add a `panic!()` in the test suite (for example in this function https://github.com/hashintel/hash/blob/925d5af5b1ac5007f54daf5e86a22200a8f3da0c/packages/engine/tests/experiment/mod.rs#L114)
3.  Run `cargo test`
4.  Confirm that no temporary files are left when `cargo test` is done
